### DISCUSSION TITLE
Streaming troubles

### DISCUSSION THREAD

**Discussion by rbagdazian** [pschatzmann/arduino-audio-tools#675] [Q&A] [answered]
I have been able to successfuly build and execute the basic-a2dp-audiokit.ino on my AI Thinker Audio Kit board and it plays perfectly. The volume control buttons work as well. I wanted to implement the a2dp input using the A2DPStream class in RX_MODE, but when I did so upon connecting my smartphone and playing the same file as before, all I get is horrible static. I played around with buffer size, buffer counts, sample rates, but nothing seems to make any difference.

I was wondering what I might be missing? Here is what my script looks like that is having all the problems:
`
#include "AudioTools.h"
#include "AudioKitSettings.h"
#include "AudioLibs/AudioKit.h"
#include "AudioLibs/AudioA2DP.h"

A2DPStream a2dp;
AudioKitStream kit;

StreamCopy copier(kit,a2dp);

void setup() {
 Serial.begin(230400);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning);

  auto cfg = a2dp.defaultConfig(RX_MODE);
  cfg.name = "BTPlayer";
  a2dp.begin(cfg);

  // setup output
  auto kcfg = kit.defaultConfig(TX_MODE);
  kcfg.sd_active = false;
  kit.begin(kcfg);

}

void loop() {
  copier.copy();
}

`
Help is much appreciated!

**Comment by rbagdazian** [comment]
I'm wondering if I should just use the  "BluetoothA2DPSink.h" class instead of the A2DPStream class since it seems like most of your examples are using it for Bluetooth input. I was trying to work toward using the StreamCopy class for handling the input to output routing and using a MultiOutput instance to route the incoming data stream to two destinations. Since I can make things work with the example I stated, is it simply a matter of sending the data to two destinations in the read callback function instead of using the MultiOutput Class, etc.? Thanks very much for your time Herr Schatzmann!

**Comment by pschatzmann** [comment, marked as answer]
Oh, you can still use the MultiOutput in the callback: so in the callback you would just write the bytes that you receive to the MultiOutput. 
But your approach works as well...

**Reply by rbagdazian** [reply]
Great, That's what I'll do. Thanks again for your kind assistance!
